### PR TITLE
Windows CI

### PR DESCRIPTION
**Pull request by tomwhite**
This is a first attempt at #247.

* The main change is to use Miniconda (so we can install msprime easily on Windows).
* I couldn't get the pre-commit GH action to work with conda, so I ended up just calling the relevant commands directly.
* Added guards to only perform some actions (linting, doc checking, etc) for one matrix entry (Ubuntu/Python 3.7) 
* I had to skip some doctest checking, since on Windows it would show numpy arrays as `(array([0, 2], dtype=int64)` not `(array([0, 2])`. I'm not sure why this is happening exactly, but I couldn't see a way in numpy to control whether the dtype was displayed or not.
* We probably want to update `docs.yml` to use the same approach for consistency.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -5,32 +5,44 @@ on: [push, pull_request]
...
+        auto-update-conda: true
+        channels: conda-forge
         python-version: ${{ matrix.python-version }}
     - name: Install dependencies
+      shell: bash -l {0}
       run: |
-        sudo apt install libgsl-dev   # Needed for msprime < 1.0. Binary wheels include GSL for >= 1.0
-        python -m pip install --upgrade pip
+        conda install msprime
         pip install -r requirements.txt -r requirements-dev.txt
```
Comment (ravwojdyla) [inline .github/workflows/build.yml:26]: Should we install these also via conda?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -5,32 +5,44 @@ on: [push, pull_request]
...
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
+    - name: Set up Miniconda with Python version ${{ matrix.python-version }}
+      uses: goanpeca/setup-miniconda@v1
       with:
+        auto-update-conda: true
+        channels: conda-forge
         python-version: ${{ matrix.python-version }}
     - name: Install dependencies
+      shell: bash -l {0}
```
Comment (ravwojdyla) [inline .github/workflows/build.yml:23]: Why is this necessary?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -5,32 +5,44 @@ on: [push, pull_request]
...
-        sudo apt install libgsl-dev   # Needed for msprime < 1.0. Binary wheels include GSL for >= 1.0
-        python -m pip install --upgrade pip
+        conda install msprime
         pip install -r requirements.txt -r requirements-dev.txt
     - name: Run pre-commit
-      uses: pre-commit/action@v2.0.0
+      if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.7'
+      shell: bash -l {0}
+      run: |
+        pre-commit install
```
Comment (ravwojdyla) [inline .github/workflows/build.yml:31]: So this is not using cache for pre commit anymore? ` FYI pre-commit/action` comes with cache built in

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -5,32 +5,44 @@ on: [push, pull_request]
...
     - name: Install dependencies
+      shell: bash -l {0}
       run: |
-        sudo apt install libgsl-dev   # Needed for msprime < 1.0. Binary wheels include GSL for >= 1.0
-        python -m pip install --upgrade pip
+        conda install msprime
         pip install -r requirements.txt -r requirements-dev.txt
     - name: Run pre-commit
-      uses: pre-commit/action@v2.0.0
+      if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.7'
```
Comment (ravwojdyla) [inline .github/workflows/build.yml:28]: Is there no better way to do this than having it repeated in each step?

**Review by ravwojdyla — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 2 above)
Comment (tomwhite) [reply .github/workflows/build.yml:23]: It's to ensure the conda env is activated. See https://github.com/marketplace/actions/setup-miniconda#important

I can add a comment to document this.

File: .github/workflows/build.yml (hunk 4 above)
Comment (tomwhite) [reply .github/workflows/build.yml:28]: Good question!

File: .github/workflows/build.yml (hunk 3 above)
Comment (tomwhite) [reply .github/workflows/build.yml:31]: Here's an example of a run where it failed: https://github.com/tomwhite/sgkit/runs/1121992680?check_suite_focus=true

```
##[error]Unable to locate executable file: pre-commit. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also check the file mode to verify the file is executable.
```

File: .github/workflows/build.yml (hunk 1 above)
Comment (tomwhite) [reply .github/workflows/build.yml:26]: I'm not sure. What's the downside of doing it like this?

Comment (ravwojdyla) [reply .github/workflows/build.yml:26]: @tomwhite not sure, but if we already use conda, why not use it all the way?

File: .github/workflows/build.yml (hunk 2 above)
Comment (ravwojdyla) [reply .github/workflows/build.yml:23]: Oh, this is why internally we use:

```
    - uses: s-weigand/setup-conda@v1
      with:
        python-version: 3.8
        conda-channels: conda-forge
```

that makes it easier to use the `base` env.

File: .github/workflows/build.yml (hunk 1 above)
Comment (jeromekelleher) [reply .github/workflows/build.yml:26]: I'd be in favour of using conda by default - it's good to test out CI on the conda infrastructure.

**Comment by jeromekelleher** [comment]
My take would be that it'd be better to write a specific workflow for windows/conda and leave the current Ubuntu workflows as they are. In practise, it never works having complicated stuff work across all three platforms. Having a specific workflow where we just do the bare minimum to make the unit tests work on Windows will make things much easier.

**Comment by tomwhite** [comment]
Thanks for all the feedback. Hows does this look now?

File: .github/workflows/windows.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
+name: Windows build
```
Comment (ravwojdyla) [inline .github/workflows/windows.yml:1]: Do you want to add this as a badge? I assume so, and if we do, this name will appear on the badge, is this the name we want (maybe just `Windows`)?

File: .github/workflows/windows.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+on: [push, pull_request]
+
+jobs:
+  build:
```
Comment (ravwojdyla) [inline .github/workflows/windows.yml:6]: Do we need to update `mergify` config (we might want to rename `build` to say `win_build` to avoid collision)?

**Review by ravwojdyla — CHANGES_REQUESTED** [review]

File: .github/workflows/windows.yml (hunk 5 above)
Comment (jeromekelleher) [reply .github/workflows/windows.yml:1]: Be nice to have the badge.

Comment (tomwhite) [reply .github/workflows/windows.yml]: Fixed in https://github.com/pystatgen/sgkit/pull/269/commits/5314a835299b1612a0928e0c33eeb28055afdcfb

File: .github/workflows/windows.yml (hunk 6 above)
Comment (tomwhite) [reply .github/workflows/windows.yml]: Fixed in https://github.com/pystatgen/sgkit/pull/269/commits/5314a835299b1612a0928e0c33eeb28055afdcfb

**Review by ravwojdyla — APPROVED** [review]

**Comment by jeromekelleher** [comment]
LGTM - ready to merge?

**Review by jeromekelleher — APPROVED** [review]
